src/beast/beast/asio/streambuf.h

```diff
@@ -442,6 +442,7 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
                     ++pos;
                     break;
                 }
+                out_end_ = pos->size();
```

Comment:
If `pos` does not point to the last element then in theory yes. This would happen if blocks were allocated at different sizes. The code was written to support it, but I do not believe that use-case is exercised (that's why `block_size_` is a class member).